src/batchview.jl

```diff
@@ -134,26 +134,45 @@ Base.@propagate_inbounds function getobs(A::BatchView)
     return _getbatch(A, 1:numobs(A.data))
 end
 
-Base.@propagate_inbounds function Base.getindex(A::BatchView, i::Int)
-    obsindices = _batchrange(A, i)
+Base.@propagate_inbounds function Base.getindex(A::BatchView, i)
+    obsindices = _batchindexes(A, i)
     _getbatch(A, obsindices)
 end
 
-Base.@propagate_inbounds function Base.getindex(A::BatchView, is::AbstractVector)
-    obsindices = union((_batchrange(A, i) for i in is)...)::Vector{Int}
-    _getbatch(A, obsindices)
-end
-
-function _getbatch(A::BatchView{TElem, TData, Val{true}}, obsindices) where {TElem, TData}
+function _getbatch(A::BatchView{<:Any, <:Any, Val{true}}, obsindices)
     batch([getobs(A.data, i) for i in obsindices])
 end
-function _getbatch(A::BatchView{TElem, TData, Val{false}}, obsindices) where {TElem, TData}
+function _getbatch(A::BatchView{<:Any, <:Any, Val{false}}, obsindices)
     return [getobs(A.data, i) for i in obsindices]
 end
-function _getbatch(A::BatchView{TElem, TData, Val{nothing}}, obsindices) where {TElem, TData}
+function _getbatch(A::BatchView{<:Any, <:Any, Val{nothing}}, obsindices)
     getobs(A.data, obsindices)
 end
 
+function getobs!(buffer, A::BatchView, i)
+    obsindices = _batchindexes(A, i)
+    return _getbatch!(buffer, A, obsindices)
+end
+
+function _getbatch!(buffer, A::BatchView{<:Any, <:Any, Val{nothing}}, obsindices)
+    return getobs!(buffer, A.data, obsindices)
+end
+
+# This collate=true specialization doesn't seem to be particularly useful, use collate=nothing instead.
+function _getbatch!(buffer, A::BatchView{<:Any, <:Any, Val{true}}, obsindices)
+    for (i, idx) in enumerate(obsindices)
+        getobs!(buffer[i], A.data, idx)
```

Comment:
I don't understand why this isn't 
```suggestion
        getobs!(buffer, A.data, idx)
```
Can you explain how it works to me briefly?

Reply:
When `collate == false`, the minibatch returned is 
```julia
[getobs(A.data, idx) for idx in obsindexes]
``` 
so in this context the buffer has to be used for the single observation. We have to use a different buffer for all observations in the mini-batch though, therefore the mini-batch buffer has to be a vector of single-observation-buffers of the length of the batchsize:
```julia
[getobs!(buffer[i], A.data, idx) for (i, idx) in for enumerate(obsindexes)]
```